src/alphabets/latin.rs

```diff
@@ -41,88 +44,128 @@ const VIE: &str =
     "abcdefghijklmnopqrstuvwxyzàáâãèéêìíòóôõùúýăđĩũơưạảấầẩẫậắằẳẵặẹẻẽếềểễệỉịọỏốồổỗộớờởỡợụủứừửữựỳỵỷỹ";
 const ZUL: &str = "abcdefghijklmnopqrstuvwxyz";
 
-fn get_lang_chars(lang: Lang) -> Vec<char> {
-    let alphabet = match lang {
-        Lang::Afr => AFR,
-        Lang::Aka => AKA,
-        Lang::Aze => AZE,
-        Lang::Cat => CAT,
-        Lang::Ces => CES,
-        Lang::Dan => DAN,
-        Lang::Deu => DEU,
-        Lang::Eng => ENG,
-        Lang::Epo => EPO,
-        Lang::Est => EST,
-        Lang::Fin => FIN,
-        Lang::Fra => FRA,
-        Lang::Hrv => HRV,
-        Lang::Hun => HUN,
-        Lang::Ind => IND,
-        Lang::Ita => ITA,
-        Lang::Jav => JAV,
-        Lang::Lat => LAT,
-        Lang::Lav => LAV,
-        Lang::Lit => LIT,
-        Lang::Nld => NLD,
-        Lang::Nob => NOB,
-        Lang::Pol => POL,
-        Lang::Por => POR,
-        Lang::Ron => RON,
-        Lang::Slk => SLK,
-        Lang::Slv => SLV,
-        Lang::Sna => SNA,
-        Lang::Spa => SPA,
-        Lang::Swe => SWE,
-        Lang::Tgl => TGL,
-        Lang::Tuk => TUK,
-        Lang::Tur => TUR,
-        Lang::Uzb => UZB,
-        Lang::Vie => VIE,
-        Lang::Zul => ZUL,
-
-        _ => panic!("No alphabet for {}", lang),
-    };
-    alphabet.chars().collect()
-}
+const LATIN_ALPHABETS: &[(Lang, &str)] = &[
+    (Lang::Afr, AFR),
+    (Lang::Aka, AKA),
+    (Lang::Aze, AZE),
+    (Lang::Cat, CAT),
+    (Lang::Ces, CES),
+    (Lang::Dan, DAN),
+    (Lang::Deu, DEU),
+    (Lang::Eng, ENG),
+    (Lang::Epo, EPO),
+    (Lang::Est, EST),
+    (Lang::Fin, FIN),
+    (Lang::Fra, FRA),
+    (Lang::Hrv, HRV),
+    (Lang::Hun, HUN),
+    (Lang::Ind, IND),
+    (Lang::Ita, ITA),
+    (Lang::Jav, JAV),
+    (Lang::Lat, LAT),
+    (Lang::Lav, LAV),
+    (Lang::Lit, LIT),
+    (Lang::Nld, NLD),
+    (Lang::Nob, NOB),
+    (Lang::Pol, POL),
+    (Lang::Por, POR),
+    (Lang::Ron, RON),
+    (Lang::Slk, SLK),
+    (Lang::Slv, SLV),
+    (Lang::Sna, SNA),
+    (Lang::Spa, SPA),
+    (Lang::Swe, SWE),
+    (Lang::Tgl, TGL),
+    (Lang::Tuk, TUK),
+    (Lang::Tur, TUR),
+    (Lang::Uzb, UZB),
+    (Lang::Vie, VIE),
+    (Lang::Zul, ZUL),
+];
+
+/// Inverted map binding a character to a set of languages.
+pub static ALPHABET_LANG_MAP: Lazy<(Vec<char>, Vec<Vec<Lang>>)> = Lazy::new(|| {
```

Comment:
Why not keep only one vec here? Is it slower?
```suggestion
/// Inverted map binding a character to a set of languages.
pub static ALPHABET_LANG_MAP: Lazy<(Vec<(char, Vec<Lang>)>)> = Lazy::new(|| {
```

Reply:
I don't use them at the same time.
`chars` is used in the first loop to compute char scores based on their occurrence in the text.
`langs` is used in the second loop to sum char scores into lang scores.
keeping them merged would just add noisy tuples with anonymous variables everywhere.

Reply:
It took me a quite a while to understand why this data structure the way it is..
The binding through the position in the vectors is not very obvious. Would you mind adding a comment on that? (in a separate PR, this one I guess will be merged today)